letsencrypt/cli.py

```diff
@@ -631,25 +642,32 @@ def __init__(self, args, plugins):
         #print self.visible_topics
         self.groups = {}  # elements are added by .add_group()
 
-    def preprocess_args(self, args):
-        """Work around some limitations in argparse.
+    def parse_args(self):
+        """Parses command line arguments and returns the result.
+
+        :returns: parsed command line arguments
+        :rtype: argparse.Namespace
 
-        Currently: add the default verb "run" as a default, and ensure that the
-        subcommand / verb comes last.
         """
-        if "-h" in args or "--help" in args:
+        parsed_args = self.parser.parse_args(self.args)
+        parsed_args.func = self.VERBS[self.verb]
+
+        return parsed_args
+
+    def determine_verb(self):
+        """Determines the verb/subcommand provided by the user."""
```

Comment:
I recommend leaving in a comment or docstring here to say that we're preprocessing to work around argparse limiations. Otherwise that might be quite confusing to future readers.